bindings/python/dlite-entity-python.i

```diff
@@ -67,6 +67,14 @@ class Metadata(Instance):
             return lst[0]
         raise DLiteError(f"Metadata {self.uri} has no such property: {name}")
 
+    def dimnames(self):
+        """Returns a list of all dimension names in this metadata."""
+        return [d.name for d in self.properties['dimensions']]
+
+    def propnames(self):
+        """Returns a list of all property names in this metadata."""
+        return [p.name for p in self.properties['properties']]
```

Comment:
Throw an informative error in the unlikely event that properties does not contain 'properties'